### PR TITLE
feat/OIDF-51 - Implement Persistence Module

### PR DESCRIPTION
**Pull request by jcmelati**

### CODE REVIEW

File: modules/persistence/src/commonMain/resources/db/migration/1.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    id  SERIAL PRIMARY KEY,
+    username VARCHAR(255) UNIQUE NOT NULL,
+    created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    deleted_at TIMESTAMP
+);
+
+CREATE INDEX account_username_index ON account (username);
+
+INSERT INTO account (username) VALUES ('root');
```
Comment (maikel-maas) [inline modules/persistence/src/commonMain/resources/db/migration/1.sql:11]: Add trailing new line

**Review by maikel-maas — CHANGES_REQUESTED** [review]
Left some remarks

File: modules/persistence/src/commonMain/resources/db/migration/1.sql (hunk 1 above)
Comment (jcmelati) [reply modules/persistence/src/commonMain/resources/db/migration/1.sql:11]: This is auto generated.

File: .env [hunk 2]
```diff
@@ -1,4 +1,4 @@
-DATASOURCE_URL=jdbc:postgresql://localhost:5432/openid-federation-db
-DATASOURCE_USER=openid-federation-db-user
-DATASOURCE_PASSWORD=openid-federation-db-password
+DATASOURCE_URL=jdbc:postgresql://localhost:5432/openid-federation-db
+DATASOURCE_USER=openid-federation-db-user
+DATASOURCE_PASSWORD=openid-federation-db-password
 DATASOURCE_DB=openid-federation-db
```
[the remark this answers is not in the document]
Comment (jcmelati) [reply .env]: fixed

File: modules/persistence/src/commonMain/resources/db/migration/1.sql (hunk 1 above)
Comment (nklomp) [reply modules/persistence/src/commonMain/resources/db/migration/1.sql:11]: Also thr trailing newlines are not that big of a thing. I rather have people focus on real issues with the code I'm a review

**Review by maikel-maas — APPROVED** [review]
LGTM

**Comment by nklomp** [comment]
Please do not merge yet. Imo the service layer needs to be decoupled into its own module, as all of that is still entirely KMP. Then other languages only have to hookup their API solution of choice to the service layer

**Comment by jcmelati** [comment]
@nklomp implemented services module as per the message above

**Pull request merged**